buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy

```diff
@@ -168,10 +168,6 @@ class PrecommitTasks {
             classpath = project.files { sourceSet.runtimeClasspath.plus(sourceSet.compileClasspath) }
 
             targetCompatibility = BuildParams.runtimeJavaVersion.majorVersion
-            if (BuildParams.runtimeJavaVersion > JavaVersion.VERSION_13) {
```

Comment:
We test with JDK 15. Should we still check to cap this at 14?

Reply:
Yep you are right. I added this back using Java 15. We should rethink using runtime java here. It isn't necessary; we only need the target version for each sourceset we are running against.

Reply:
I agree. What we _should_ be using is `minimumRuntimeVersion` since that is what we use as our target compatibility version and pass via `-release` to `java`. Just because we test with a new runtime version doesn't in any way change how we _compile_.

Reply:
It does change for MRJARs, which is why I think we should set it to the compile version for the specific sourceset.

Reply:
Exactly, thanks for this @breskeby. The target version passed to forbiddenapis should exactly be the same like passed to javac. By default the plugin also sets this up from the targetVersion property used also by javac (project.targetVersion: https://docs.gradle.org/current/userguide/java_plugin.html)

For MR-JARS this is not yet ideal in Gradle. IMHO, Gradle should define the target/sourceVersion on the sourceSet and not on the project. If this would be the case, you would set the sourceSet's targetVersion and forbiddenapis would automatically take it.

If there is any progress with changes to Gradle's support for MR-JARs, please open an issue in forbiddenapis, to adapt the defaults in plugin-init.gradle.